streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -800,35 +800,39 @@ record = null;
                 record = null;
                 throw new TaskCorruptedException(Collections.singleton(id));
             }
+        } catch (final FailedProcessingException failedProcessingException) {
+            // Do not keep the failed processing exception in the stack trace
+            handleException(new Exception(failedProcessingException.getCause()));
         } catch (final StreamsException exception) {
             record = null;
             throw exception;
         } catch (final RuntimeException e) {
```

Comment:
Didn't we agree on using `Exception` here instead of `RuntimeException` to be consistent with other handlers?

Reply:
Fixed